app/src/main/java/com/spascoding/taskycourse/feature_auth/data/repository/AuthRepositoryImpl.kt

```diff
@@ -54,4 +60,63 @@ class AuthRepositoryImpl @Inject constructor(
         return response.isSuccessful
     }
 
+    //Utils
+
+    private suspend fun <T> performRequest(
+        request: suspend () -> Response<T>,
+        onSuccess: suspend (T) -> Unit = {}
+    ): Result<T, DataError.Remote> {
+        return try {
+            val response = request()
+            if (response.isSuccessful) {
+                response.body()?.let {
+                    onSuccess(it)
+                    Result.Success(it)
+                } ?: Result.Error(DataError.Remote.UNKNOWN)
+            } else {
+                //How I can return error massage from server and it needs to do this?
+//                val errorResponse = response.errorBody()?.string()
+//                val errorMessage = handleErrorMassage(errorResponse)
+//                Result.Error(TextError(errorMessage))
```

Comment:
What you wrote there looks good to me, that's at least the right direction. If you get specific errors with that, let me know. However, in this project you shouldn't need that since there are unique messages per HTTP code returned. The problem with server side messages is that they can't be localized correctly.